### PR TITLE
Make max_insert_delayed_streams_for_parallel_write actually work

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by alesapin** [comment]
cc @azat

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit a11e67d4aae4433dd0f3d8ee46ba40e1cd73fdd5 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60079/a11e67d4aae4433dd0f3d8ee46ba40e1cd73fdd5/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60079/a11e67d4aae4433dd0f3d8ee46ba40e1cd73fdd5/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp [hunk 1]
```diff
@@ -323,6 +323,9 @@ void ReplicatedMergeTreeSinkImpl<async_insert>::consume(Chunk chunk)
         if (!temp_part.part)
             continue;
 
+        if (!support_parallel_write && temp_part.part->getDataPartStorage().supportParallelWrite())
```
Comment (KochetovNicolai) [inline src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp:326]: I also expect something like
```
size_t max_insert_delayed_streams_for_parallel_write = DEFAULT_DELAYED_STREAMS_FOR_PARALLEL_WRITE;
if (settings.max_insert_delayed_streams_for_parallel_write.changed)
    max_insert_delayed_streams_for_parallel_write = settings.max_insert_delayed_streams_for_parallel_write;


if (!support_parallel_write)
    max_insert_delayed_streams_for_parallel_write = 0;
```
below

**Review by KochetovNicolai — APPROVED** [review]

**Comment by azat** [comment]
@alesapin you mean that it works differently when the default value of `max_insert_delayed_streams_for_parallel_write` is different from 0?

**Comment by alesapin** [comment]
@azat there were just now logic for support_parallel_write in replicated merge tree. It was always false.

**Pull request merged**